inference-engine/src/transformations/src/transformations/common_optimizations/weights_dequantize_to_fake_quantize.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/weights_dequantize_to_fake_quantize.hpp"
+
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/rt_info.hpp>
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::WeightsDequantizeToFakeQuantize, "WeightsDequantizeToFakeQuantize", 0);
+
+ngraph::pass::WeightsDequantizeToFakeQuantize::WeightsDequantizeToFakeQuantize() {
+    MATCHER_SCOPE(WeightsDequantizeToFakeQuantize);
+
+    const auto weights = ngraph::pattern::wrap_type<ngraph::opset6::Constant>();
+    const auto convert = ngraph::pattern::wrap_type<ngraph::opset6::Convert>({weights});
+    const auto sub_c = ngraph::pattern::wrap_type<ngraph::opset6::Constant>();
+    const auto sub = ngraph::pattern::wrap_type<ngraph::opset6::Subtract>({convert, sub_c});
+
+    const auto sub_or_convert = std::make_shared<pattern::op::Or>(OutputVector{convert, sub});
+
+    const auto mul_c = ngraph::pattern::wrap_type<ngraph::opset6::Constant>();
+    const auto mul = ngraph::pattern::wrap_type<ngraph::opset6::Multiply>({sub_or_convert, mul_c});
+
+    ngraph::matcher_pass_callback callback;
+    callback = [=](ngraph::pattern::Matcher &m) {
+        auto pattern_map = m.get_pattern_map();
+
+        const auto &weights_node = as_type_ptr<opset6::Constant>(pattern_map.at(weights));
+        const auto &convert_node = pattern_map.at(convert);
+        const auto &multiply_node = pattern_map.at(mul);
+        const auto &scale_node = pattern_map.at(mul_c);
+        if (!weights_node || !convert_node || !multiply_node || !scale_node) {
+            return false;
+        }
+
+        const auto *data = weights_node->get_data_ptr<int8_t>();
```

Comment:
we need to check that data has int8_t type. otherwise you will get wrong values as get_data_ptr doesn't cast values.